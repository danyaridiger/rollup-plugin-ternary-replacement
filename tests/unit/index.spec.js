import createOutput from "../tools/create-output";

describe("rollup-plugin-ternary-replacement", () => {
  it("correctly transforms all given nodes", async () => {
    const dist = await createOutput();

    expect(dist).toMatchSnapshot();
  });


  it("correctly excludes files from transform", async () => {
    const dist = await createOutput({ exclude: ["exclude"] });

    expect(dist).toMatchSnapshot();
  });


  it("correctly excludes files from transform by extentions", async () => {
    const dist = await createOutput({ excludeExtentions: ["js"] });

    expect(dist).toMatchSnapshot();
  });


  it("correctly transforms only non null merges", async () => {
    const dist = await createOutput({ nonNullMergesOnly: true });

    expect(dist).toMatchSnapshot();
  });
  

  it("correctly transforms only logical assignments", async () => {
    const dist = await createOutput({ assignmentsOnly: true });

    expect(dist).toMatchSnapshot();
  });
});