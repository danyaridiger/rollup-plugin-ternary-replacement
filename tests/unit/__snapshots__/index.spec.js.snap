// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-ternary-replacement correctly excludes files from transform 1`] = `
"/**
 * Function that imitates module behavior for tests.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable2;

  return testVariable1;
};

export { testFunction as default };

/**
 * Function that imitates module behavior for tests.
 * Can be excluded from conversion using "exclude" option.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable1 ?? testVariable2;

  return testVariable1;
};

export { testFunction as default };
"
`;

exports[`rollup-plugin-ternary-replacement correctly excludes files from transform by extentions 1`] = `
"/**
 * Function that imitates module behavior for tests.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  let testVariable3 = 120;
  let testVariable4 = 130;
  let testVariable5 = 140;
  
  testVariable1 = testVariable2;
  testVariable2 &&= testVariable3;
  testVariable3 ||= testVariable4;
  testVariable4 ??= testVariable5;

  return testVariable1;
};

export { testFunction as default };

/**
 * Function that imitates module behavior for tests.
 * Can be excluded from conversion using "exclude" option.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable1 ?? testVariable2;

  return testVariable1;
};

export { testFunction as default };
"
`;

exports[`rollup-plugin-ternary-replacement correctly transforms all given nodes 1`] = `
"/**
 * Function that imitates module behavior for tests.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable2;

  return testVariable1;
};

export { testFunction as default };

/**
 * Function that imitates module behavior for tests.
 * Can be excluded from conversion using "exclude" option.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable1 !== null && testVariable1 !== undefined ? testVariable1 : testVariable2;

  return testVariable1;
};

export { testFunction as default };
"
`;

exports[`rollup-plugin-ternary-replacement correctly transforms only logical assignments 1`] = `
"/**
 * Function that imitates module behavior for tests.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable2;

  return testVariable1;
};

export { testFunction as default };

/**
 * Function that imitates module behavior for tests.
 * Can be excluded from conversion using "exclude" option.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable1 ?? testVariable2;

  return testVariable1;
};

export { testFunction as default };
"
`;

exports[`rollup-plugin-ternary-replacement correctly transforms only non null merges 1`] = `
"/**
 * Function that imitates module behavior for tests.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  let testVariable3 = 120;
  let testVariable4 = 130;
  let testVariable5 = 140;
  
  testVariable1 = testVariable2;
  testVariable2 &&= testVariable3;
  testVariable3 ||= testVariable4;
  testVariable4 ??= testVariable5;

  return testVariable1;
};

export { testFunction as default };

/**
 * Function that imitates module behavior for tests.
 * Can be excluded from conversion using "exclude" option.
 * @function
 * @returns {number} - result
 */
const testFunction = () => {
  let testVariable1 = 100;
  let testVariable2 = 110;
  
  testVariable1 = testVariable1 !== null && testVariable1 !== undefined ? testVariable1 : testVariable2;

  return testVariable1;
};

export { testFunction as default };
"
`;
